dev/com.ibm.ws.jpa_spec10_locking_fat/test-applications/EntityLocking/src/com/ibm/ws/jpa/fvt/entitylocking20/testlogic/EntityLocking20TestLogic.java

```diff
@@ -0,0 +1,1268 @@
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.jpa.fvt.entitylocking20.testlogic;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityTransaction;
+import javax.persistence.LockModeType;
+import javax.persistence.OptimisticLockException;
+
+import org.junit.Assert;
+
+import com.ibm.ws.jpa.fvt.entitylocking20.entities.LockEntityA;
+import com.ibm.ws.testtooling.testinfo.TestExecutionContext;
+import com.ibm.ws.testtooling.testlogic.AbstractTestLogic;
+import com.ibm.ws.testtooling.vehicle.resources.JPAResource;
+import com.ibm.ws.testtooling.vehicle.resources.TestExecutionResources;
+
+/**
+ *
+ */
+public class EntityLocking20TestLogic extends AbstractTestLogic {
+
+    /**
+     *
+     *
+     * @param testExecCtx
+     * @param testExecResources
+     * @param managedComponentObject
+     */
+    public void testReadLock001(
+                                TestExecutionContext testExecCtx,
+                                TestExecutionResources testExecResources,
+                                Object managedComponentObject) throws Throwable {
+        final String testName = getTestName();
+
+        // Verify parameters
+        if (testExecCtx == null || testExecResources == null) {
+            Assert.fail(testName + ": Missing context and/or resources.  Cannot execute the test.");
+            return;
+        }
+
+        // Fetch JPA Resources
+        JPAResource jpaResource = testExecResources.getJpaResourceMap().get("test-jpa-resource");
+        if (jpaResource == null) {
+            Assert.fail("Missing JPAResource 'test-jpa-resource').  Cannot execute the test.");
+            return;
+        }
+
+        AbstractClientB cliBTL = null;
+        // Execute Test Case
+        try {
+            System.out.println(testName + ": Begin");
+
+            EntityManager em = jpaResource.getEm();
+            em.clear();
+
+            System.out.println("Beginning new transaction...");
+            jpaResource.getTj().beginTransaction();
+
+            if (jpaResource.getTj().isApplicationManaged()) {
+                System.out.println("Joining entitymanager to JTA transaction...");
+                jpaResource.getEm().joinTransaction();
+            }
+
+            System.out.println("Finding LockEntityA(id=1) with Optimistic Read locking...");
+            LockEntityA entityA = jpaResource.getEm().find(LockEntityA.class, 1, LockModeType.READ);
```

Comment:
This test appears to be testing a JPA 2.0 API feature, is that correct? Should this test be moved to the jpa20 spec FAT?

Reply:
Fixed the version umbers.